inc/composer/class-command.php

```diff
@@ -781,6 +782,42 @@ protected function get_project_subdomain() : string {
 		return preg_replace( '/[^A-Za-z0-9\-\_]/', '', $project_name );
 	}
 
+	/**
+	 * Run a prepared process command for various versions of Symfony Console.
+	 *
+	 * Console v5+ requires an array for the command.
+	 * Console v1-3 only supports strings.
+	 *
+	 * @param mixed ...$args Args to pass to Process.
+	 * @return Process
+	 */
+	protected function process( ...$args ) : Process {
+		if ( version_compare( $this->get_composer_version(), '2.3', '>=' ) && ! is_array( $args[0] ) ) {
+			$args[0] = explode( ' ', $args[0] );
+		}
+
+		return new Process( ...$args );
+	}
+
+	/**
+	 * Get the composer executable version.
+	 *
+	 * @return string|null
+	 */
+	protected function get_composer_version() : ?string {
+		static $version;
+
+		if ( $version ) {
+			return $version;
+		}
+
+		preg_match( '/^Composer ([\d\.-]+)/', exec( 'composer --version' ), $composer_version );
```

Comment:
Could this use `Composer::getVersion()` instead since we're already running in Composer context?

Reply:
Wasn't aware that was a thing! Ta